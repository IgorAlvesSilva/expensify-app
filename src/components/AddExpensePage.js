import React from 'react';
import ExpenseFrom from './ExpenseForm';
import {connect} from 'react-redux'
import { startAddExpense } from '../actions/expenses'


export class AddExpensePage extends React.Component {
  onSubmit = (expense => {
    this.props.startAddExpense(expense);
    this.props.history.push("/")
  })

  render() {
    return (
      <div>
      This is from my add expense component
      <ExpenseFrom 
        onSubmit={this.onSubmit}
      />
    </div>    
    )}}


const mapDispatchToProps = (dispatch) => {
  return {
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
  };
}
export default connect(undefined, mapDispatchToProps)(AddExpensePage);
