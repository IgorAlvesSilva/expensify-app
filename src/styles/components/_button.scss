.button {
    border: none;
    color: white;
    background-color: $blue;
    padding: $s-size;
    font-weight: 300;
    font-size: $font-size-large;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888888;
}