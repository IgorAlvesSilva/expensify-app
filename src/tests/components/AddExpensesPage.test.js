import React from "react";
import {AddExpensePage} from "../../components/AddExpensePage";
import { shallow } from 'enzyme'
import ExpenseForm from "../../components/ExpenseForm";
import expenses from '../fixtures/expenses'

let startAddExpense, history, wrapper;

beforeEach(() => {
    startAddExpense = jest.fn();
    history = {push: jest.fn()}
    wrapper = shallow(<AddExpensePage startAddExpense={startAddExpense} history={history} />);
});

test('should render startAddExpense correcly', () => {
    expect(wrapper).toMatchSnapshot();
});


test('should handle onSumit', () => {
    wrapper.find(ExpenseForm).prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(startAddExpense).toHaveBeenLastCalledWith(expenses[1]);
});