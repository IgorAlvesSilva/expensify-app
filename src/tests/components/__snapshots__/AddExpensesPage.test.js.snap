// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render startAddExpense correcly 1`] = `
<div>
  This is from my add expense component
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
